runtime/curl-async.cpp

```diff
@@ -14,33 +14,32 @@ class curl_exec_concurrently final : public Resumable {
   using ReturnT = Optional<string>;
 
   const double timeout_s{0.0};
-  const int request_id{0};
   int resumable_id{0};
-  std::unique_ptr<CurlRequest> request;
+  const CurlRequest request;
   std::unique_ptr<CurlResponse> response;
 
 public:
-  curl_exec_concurrently(std::unique_ptr<CurlRequest> &&request, double timeout_s) noexcept
+  curl_exec_concurrently(const CurlRequest &request, double timeout_s) noexcept
     : timeout_s(timeout_s)
-    , request_id(request->request_id)
-    , request(std::move(request)) {}
+    , request(request) {}
 
   bool run() noexcept final {
     RESUMABLE_BEGIN
-    resumable_id = vk::singleton<CurlAdaptor>::get().launch_request_resumable(std::move(request));
+    resumable_id = vk::singleton<CurlAdaptor>::get().launch_request_resumable(request);
     response = f$wait<std::unique_ptr<CurlResponse>, false>(resumable_id, timeout_s);
     TRY_WAIT(curl_exec_concurrently_label, response, std::unique_ptr<CurlResponse>);
-    vk::singleton<CurlAdaptor>::get().finish_request(request_id);
+    vk::singleton<CurlAdaptor>::get().finish_request(request);
     RETURN(response ? response->response : ReturnT{false});
     RESUMABLE_END
   }
 };
 } // namespace curl_async
 
 Optional<string> f$curl_exec_concurrently(curl_easy easy_id, double timeout_s) {
```

Comment:
let's mark it as `noexcept` just in case

Reply:
Let me leave as is pls